Modules/Forum/classes/class.ilForumTopic.php

```diff
@@ -565,40 +565,38 @@ public function movePosts($old_obj_id, $old_pk, $new_obj_id, $new_pk)
 				}
 			}
 
-			$this->db->lockTables(
-				array(
-					0 => array('name' => 'frm_user_read',     'type' => ilDBConstants::LOCK_WRITE),
-					1 => array('name' => 'frm_thread_access', 'type' => ilDBConstants::LOCK_WRITE)
-				)
-			);
-
-			$this->db->manipulateF('
+			$ilAtomQuery = new ilAtomQuery($ilDB);
```

Comment:
Well done @chfsx, looks easy to use. I also checked the code in **\ilBuddySystemRelationRepository::save** which could be easily changed as well.

@bheyser: This approach is (of course) a little different compared to your **\ilTestProcessLocker**. I only found two usages of the respective methods. In my opinion your lock mechanism has to require a closure (instead of calling lock/unlock in the consuming code), and in case of database locking (set in the assessment administration) it should pass this closure to **\ilAtomQuery**.